src/Fixer/Whitespace/UnionTypesSpacesFixer.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class UnionTypesSpacesFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'A single space or none should be around union type operator.',
+            [
+                new CodeSample(
+                    "<?php\ntry\n{\n    new Foo();\n} catch (ErrorA | ErrorB \$e) {\necho'error';}\n"
+                ),
+                new CodeSample(
+                    "<?php\ntry\n{\n    new Foo();\n} catch (ErrorA|ErrorB \$e) {\necho'error';}\n",
+                    ['space' => 'single']
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction foo(int | string \$x)\n{\n}\n",
+                    new VersionSpecification(80000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(CT::T_TYPE_ALTERNATION);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run after BinaryOperatorSpacesFixer.
+     */
+    public function getPriority(): int
+    {
+        return -33;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('space', 'spacing to apply around union type operator.'))
+                ->setAllowedValues(['none', 'single'])
+                ->setDefault('none')
```

Comment:
Isn't `single` the most widespread code style?

Reply:
Is it? Both [RFC](https://wiki.php.net/rfc/union_types_v2) and @nicolas-grekas [request](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5733) are for no space.

Reply:
clearly `none` is the most widespread cs on this, see even the phpdoc